clang/include/clang/Basic/WebAssemblyReferenceTypes.def

```diff
@@ -36,5 +36,7 @@
 
 WASM_REF_TYPE("__externref_t", "externref_t", WasmExternRef, WasmExternRefTy, 10)
 
+WASM_REF_TYPE("__non_null_externref_t", "non_null_externref_t", WasmNonNullExternRef, WasmNonNullExternRefTy, 10)
```

Comment:
Maybe this should be 11 or something... So that when we constraint the types behaviour we can differentiate it from (the nullable) externref.